semantics/language/execution/check-loc.k

```diff
@@ -70,112 +80,101 @@ module C-CHECK-LOC
-     //      ...</k>
-     //      <modified> Modified:Map </modified>
-     //      requires notBool (Tag in keys(Modified))
-     //      [structural]
-     // rule <k> checkRestrict(Loc:SymLoc, _, _,
-     //                (SetItem(basedOn(Base:SymBase, Tag:RestrictTag)) => .Set) _,
-     //                (.Set => SetItem(basedOn(Base, Tag))) _)
-     //      ...</k>
-     //      <modified>... Tag |-> map(Modified:Map) ...</modified>
-     //      requires Loc in keys(Modified)
-     //      [structural]
-     // // TODO(chathhorn): merging; might need work.
-     // rule <k> checkRestrict(Loc:SymLoc, _, _,
-     //                (SetItem(basedOn(_, Tag:RestrictTag)) => .Set) _, _)
-     //      ...</k>
-     //      <modified>... Tag |-> map(Modified:Map) ...</modified>
-     //      requires notBool (Loc in keys(Modified))
-     //      [structural]
-     // rule <k> checkRestrict(_, _, _,
-     //                (SetItem(basedOn(_, Tag:RestrictTag)) => .Set) _, _)
-     //      ...</k>
-     //      <modified> Modified:Map </modified>
-     //      requires notBool (Tag in keys(Modified))
-     //      [structural]
-     // rule checkRestrict(_, .Set, Bases:Set, .Set, Bases':Set)
-     //           => checkRestrict(Bases Bases', Bases, Bases')
-     //      [structural]
-
-     // // Finally, do the actual checks.
-     // syntax KItem ::= checkRestrict(Set, Set, Set)
-     // rule (.K => checkRestrict'(B))
-     //      ~> checkRestrict((SetItem(B:Provenance) => .Set) _, _, _)
-     //      [structural]
-     // rule checkRestrict(.Set, Bases:Set, Bases':Set) => .
-     //      requires Bases ==Set Bases'
-     //      [structural]
-     // rule (.K => UNDEF("ECL3",
-     //           "An object which has been modified is accessed through an expression based on a restrict-qualified pointer and another lvalue not also based on said pointer.",
-     //           "6.7.3.1:4"))
-     //      ~> checkRestrict(.Set, Bases:Set, Bases':Set)
-     //      requires Bases =/=Set Bases'
-     //      [structural]
-
-     // syntax KItem ::= "checkRestrict'" "(" Provenance ")"
-     // rule <k> checkRestrict'(basedOn(Base:SymBase, _)) => .K ...</k>
-     //      <mem>... Base |-> object(T:Type, _, _) ...</mem>
-     //      requires notBool isConstType(innerType(T))
-     //      [structural]
-     // rule <k> (.K => UNDEF("ECL4",
-     //           "An object which has been modified is accessed through an expression based on a restrict-qualified pointer to a const-qualified type.",
-     //           "6.7.3.1:4"))
-     //           ~> checkRestrict'(basedOn(Base:SymBase, _))
-     //      ...</k>
-     //      <mem>... Base |-> object(T:Type, _, _) ...</mem>
-     //      requires isConstType(innerType(T))
-     //      [structural]
+     // Split out the prov to normalize the address.
+     rule checkRestrict(loc(Base:SymBase, Offset:Int))
+          => checkRestrict(loc(Base, Offset), .Set)
```

Comment:
so, this isn't an action to be taken immediately, but just a thought: maybe we should declare the following rule as a macro?
```
rule loc(Base:SymBase, Offset:Int) => loc(Base, Offset, .Set)
```

This would potentially significantly simplify the semantics in places while still allowing us to match on the 2-argument form.